toolchain/check/testdata/array/array_in_place.carbon

```diff
@@ -82,20 +82,20 @@ fn G() {
 // CHECK:STDOUT:   %v.var: ref %array_type = var v
 // CHECK:STDOUT:   %v: ref %array_type = bind_name v, %v.var
 // CHECK:STDOUT:   %F.ref.loc14_34: %F.type = name_ref F, file.%F.decl [template = constants.%F]
-// CHECK:STDOUT:   %.loc14_42.2: ref %tuple.type.2 = splice_block %.loc14_42.1 {
```

Comment:
I have not figured out how these suffix numbers come to be, but the non-recursive walk is slightly differently ordered, in that it queues the recursion step for the end instead of starting it immediately. Not sure if this is good/bad. We can insert the recursive stuff right after the current instruction, though that is still slightly different as it would be queued after the current one instead of completing before it.

Reply:
(If you would like me to figure out why they are different here let me know and I will keep looking, and try to avoid that change, it's not clear to me if this matters or what)

Reply:
TBH this looks like an improvement since the indices increase as code progresses. I suspect this is the `break;` after `SpliceBlock` in action, if that helps.

Reply:
Yes, you're right. To preserve it we'd need to re-queue the current SpliceBlock instruction first before queueing the block id, in such a way that we'd then name the current SpliceBlock instruction after it, without also redoing the block id again infinitely.

But since you like the change I will leave it.